### PR TITLE
feat(settings): A user can edit the root key name

### LINKED ISSUE #1979: Root Key doesn't have feature to edit name after creation
### Preliminary Checks

- [X] I have reviewed https://unkey.com/docs for existing features that would solve my problem

- [X] I have searched for existing feature requests: https://github.com/unkeyed/unkey/issues

- [X] This issue is not a question, general help request, or anything other than a feature request directly related to Unkey. Please ask questions in our Discord community: https://unkey.com/discord.

### Is your feature request related to a problem? Please describe.

Create a rootkey with a name. There is no way to edit the name after creation.

### Describe the solution

Allow users to edit the name after the rootkey is created

### Describe alternatives you have considered (if any)

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by mcstepp**
## What does this PR do?

Fixes #1979 

*If there is not an issue for this, please create one first. This is used to tracking purposes and also helps use understand why this PR exists*

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Chore (refactoring code, technical debt, workflow improvements)
- [ ] Enhancement (small improvements)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How should this be tested?

1. Log into Unkey
2. Go to Settings > Root Keys
3. Click the [...] next to your root key that you want to modify. If you don't have one, create one.
4. Select Details
5. Update the name of the root key in the appropriate card. You can also delete the name if you wanted. Names are optional.
6. Click Save
7. Success

https://www.loom.com/share/1d1e6113ab3a4078813d3c4c3666fb3f?sid=0b6e0c44-8e4c-4239-87a9-be5835ec84f8

### Required

- [x] Filled out the "How to test" section in this PR
- [x] Read [Contributing Guide](./CONTRIBUTING.md)
- [x] Self-reviewed my own code
- [ ] Commented on my code in hard-to-understand areas
- [x] Ran `pnpm build`
- [ ] Ran `pnpm fmt`
- [x] Checked for warnings, there are none
- [x] Removed all `console.logs`
- [x] Merged the latest changes from main onto my branch with `git pull origin main`
- [x] My changes don't cause any responsiveness issues

### Appreciated

- [x] If a UI change was made: Added a screen recording or screenshots to this PR
- [ ] Updated the Unkey Docs if changes were necessary

## Summary by CodeRabbit

- **New Features**
	- Introduced a new component for updating the name of root keys, enhancing user management of API keys.
	- Streamlined the display of key metrics by removing the key name, focusing on essential attributes like ID and creation date.

- **Bug Fixes**
	- Improved error handling and user notifications for key name updates.

### CODE REVIEW

File: apps/dashboard/lib/trpc/routers/key/updateRootKeyName.ts [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+  .mutation(async ({ input, ctx }) => {
+    const key = await db.query.keys.findFirst({
+      where: (table, { eq, isNull, and }) =>
+        and(eq(table.id, input.keyId), isNull(table.deletedAt)),
+      with: {
+        workspace: true,
+      },
+    });
+
+    if (!key || key.ownerId !== ctx.tenant.id) {
```
Comment (chronark) [inline apps/dashboard/lib/trpc/routers/key/updateRootKeyName.ts:25]: You need to check the workspace tenant is
Not owner id

**Review by chronark — CHANGES_REQUESTED** [review]

File: apps/dashboard/lib/trpc/routers/key/updateRootKeyName.ts (hunk 1 above)
Comment (perkinsjr) [reply apps/dashboard/lib/trpc/routers/key/updateRootKeyName.ts:25]: Get off the internet. I am reviewing the code

Comment (perkinsjr) [inline apps/dashboard/lib/trpc/routers/key/updateRootKeyName.ts:25]: Let's do look up the workspace and make sure the rootKey belongs to the `for_workspace_id`  so we can be 100% sure that the key belongs. 
```
const workspace = await db.query.workspaces.findFirst({
      where: (table, { and, eq, isNull }) =>
        and(eq(table.tenantId, ctx.tenant.id), isNull(table.deletedAt)),
    });
    if (!workspace) {
      throw new TRPCError({
        code: "NOT_FOUND",
        message:
          "We are unable to find the correct workspace. Please contact support using support@unkey.dev.",
      });
    }
````

Then do 

if(!key || key.forWorkspaceId !== workspace.id

**Review by perkinsjr — CHANGES_REQUESTED** [review]
For t

File: apps/dashboard/lib/trpc/routers/key/updateRootKeyName.ts (hunk 1 above)
Comment (mcstepp) [reply apps/dashboard/lib/trpc/routers/key/updateRootKeyName.ts:25]: @chronark the workspace tenant on the root key is the generic one, and will never match the `ctx.tenantId` -- it matches the tenant of the workspace associated to the root key's `for_workspaceId` 

Unless I grab the workspace that matches the key.for_workspaceId, and compare its tenantId to ctx.tenantId.

**Review by perkinsjr — APPROVED** [review]
Code looks good.

Root key updated for both orgs and users. 
Audit log displays correctly.

**Pull request merged**